app/utils/ControllerHelper.scala

```diff
@@ -137,19 +137,39 @@ object ControllerHelper {
     }
   }
 
+  def hasCompleteNominatedOfficer(responsiblePeople: Option[Seq[ResponsiblePerson]]): Boolean = {
+    responsiblePeople match {
+      case Some(rps) => ResponsiblePerson.filter(rps).exists(rp => rp.isComplete && rp.isNominatedOfficer)
```

Comment:
Will this filter out deleted entries (with a line id). I.e. soft deleted. Maybe isComplete handles this?

Reply:
filter on responsible people will filter out empty rp objects and rps with status: Deleted, lineId is not taken into account when filtering or checking for completeness in any case. So this will return only nominated officer, who is complete and not with status deleted. I don't know if this answers your question :)